### PR TITLE
Fix display init sequence for Lilygo T-Display rp2040 and add missing 8 pins

### PR DESCRIPTION
**Pull request by kreier**
The init sequence in board.c had one additional `auto_brightness` setting in the `common_hal_busdisplay_busdisplay_construct()` of the `display_init()` function that caused a one pixel noise line at the bottom of the display and a slow refresh rate from below 1 Hertz. Described in https://github.com/adafruit/circuitpython/issues/8765 

Removing this line solved the issue.

Furthermore the init sequence from the Adafruit Feather esp32s3 tft was copied to rotate the refresh of the display to "top to bottom" and have a final `rotation=0` setting.

And the 8 missing pins in the `pins.c` were added.

### CODE REVIEW

File: ports/raspberrypi/boards/lilygo_t_display_rp2040/board.c [hunk 1]
```diff
@@ -99,9 +99,9 @@ static void display_init(void) {
         bus,
         240,            // width (after rotation)
         135,            // height (after rotation)
-        53,             // column start
-        40,             // row start
-        90,             // rotation
+        40,             // column start
+        53,             // row start
+        0,              // rotation
```
Comment (jepler) [inline ports/raspberrypi/boards/lilygo_t_display_rp2040/board.c:104]: Configurations which have a "0 degree" rotation of the displayio display are almost always preferable.

File: ports/raspberrypi/boards/lilygo_t_display_rp2040/board.c [hunk 2]
```diff
@@ -116,13 +116,13 @@ static void display_init(void) {
         &pin_GPIO4,     // backlight pin
         NO_BRIGHTNESS_COMMAND,
         1.0f,           // brightness (ignored)
-        false,          // auto_brightness
```
Comment (jepler) [inline ports/raspberrypi/boards/lilygo_t_display_rp2040/board.c:119]: I'm surprised the display worked at all with the argument positions shifted like this!

**Review by jepler — DISMISSED** [review]
looks good, no HW for testing.

**Comment by kreier** [comment]
Sorry, not dismissed. But I found one error that was not obvious with B/W text but with the use of color. The D3 bit of MADCTL changes the color information. It was initially zero and therefore RGB but with the change to `0x68` we set it to one and therefore BGR. The change to `0x60` switches the order back to RGB and is consistent with the use in T-Display ESP32. So the change from the original `0xC0` only changes the orientation and refresh direction to allow for `rotation=0`

**Comment by kreier** [comment]
Sorry, my mistake. The correct value for MADCTL is `0x68`. I checked three values in different builds and the impact on the screen:

- __0x08__ is the old value used prior to this pull request. colors are fine, refresh from left to right
- __0x60__ flips the `D3` bit of MADCTL. Colors created by adafruit_display_text are not affected, RGB values submitted with the color attribute in Label render the RGB color as intended and before. But colors used as `color_palette[0]` in `displayio.Palette(1)` and applied to `displayio.TileGrid` as `pixel_shader` argument no longer represent RGB.
- __0x68__ has the correct colors and draws from top to bottom

Now I have to find a way to retract my last commit.

It looks like general consensus to have `D3` set to `1` across different boards with this display and the ST7789:

| board                                                                                                                                           | MADCTL | _MADCTL    | rotation | column start | row start | width | height |
|-------------------------------------------------------------------------------------------------------------------------------------------------|--------|------------|----------|--------------|-----------|-------|--------|
| [T-Display rp2040](https://github.com/adafruit/circuitpython/blob/main/ports/raspberrypi/boards/lilygo_t_display_rp2040/board.c)                | 0x08   | 0b00001000 |       90 |           53 |        40 |   240 |    135 |
| [T-Display ESP32 16M](https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/lilygo_ttgo_tdisplay_esp32_16m/board.c)        | 0x08   | 0b00001000 |      270 |           53 |        40 |   240 |    135 |
| [Feather ESP32-S2 with TFT](https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/adafruit_feather_esp32s2_tft/board.c)    | 0x68   | 0b01101000 |        0 |           40 |        53 |   240 |    135 |
| [Feather ESP32-S2 reverse TFT](https://github.com/adafruit/circuitpython/tree/main/ports/espressif/boards/adafruit_feather_esp32s2_reverse_tft) | 0x68   | 0b01101000 |        0 |           40 |        53 |   240 |    135 |
| [TTGO T8 ESP32-S2 ST7789](https://github.com/adafruit/circuitpython/tree/main/ports/espressif/boards/lilygo_ttgo_t8_s2_st7789)                  | 0x08   | 0b00001000 |       90 |           52 |        40 |   240 |    135 |
| [Lilygo T-Embed ESP32-S3](https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/lilygo_tembed_esp32s3/board.c)             | 0xC8   | 0b11001000 |       90 |           35 |         0 |   320 |    170 |
| [MEMENTO](https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/adafruit_esp32s3_camera/board.c)                           | 0xA0   | 0b10100000 |        0 |           80 |         0 |   240 |    240 |
| [CLUE NRF52840 Express](https://github.com/adafruit/circuitpython/blob/main/ports/nrf/boards/clue_nrf52840_express/board.c)                     | 0xA0   | 0b10100000 |        0 |           80 |         0 |   240 |    240 |
| [Lilygo T-Deck](https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/lilygo_tdeck/board.c)                                | 0x08   | 0b00001000 |      270 |            0 |         0 |   320 |    240 |
| [M5StickC PLUS (ST7735R)](https://github.com/adafruit/circuitpython/blob/main/ports/espressif/boards/m5stack_stick_c_plus/board.c)              | 0x08   | 0b00001000 |        1 |           40 |        52 |   135 |    240 |

**Review by jepler — APPROVED** [review]
Thank you for the extended testing.

**Pull request merged**